### PR TITLE
Mana Battery Changes

### LINKED ISSUE #1374: Mana batteries don't keep power when broken anymore
Since the days where the 1.6 builds of AM2 had issues with mana networks resetting, I've taken to transferring stored power to mana batteries and then stashing them until I need to draw power from them. They used to drop with a tag regarding how full they are and what sort of essence is stored.

Now when I break them and place them again, they revert to being empty.

(This happens with a dev version I built today)

### PR DESCRIPTION
**Pull request by TheIcyOne**
Mana battery can now be picked up with sneak-use, providing an itemStack which can then be placed without loss of power. Semi-fixes #1374.

Also updated ArcaneCompendium_en_US.xml to reflect this change.

### CODE REVIEW

File: src/main/java/am2/blocks/BlockManaBattery.java [hunk 1]
```diff
@@ -114,10 +125,10 @@ public void onBlockExploded(World world, int x, int y, int z, Explosion explosio
 		destroy(world, x, y, z);
 		super.onBlockExploded(world, x, y, z, explosion);
 	}
-
+
```
Comment (jjtParadox) [inline src/main/java/am2/blocks/BlockManaBattery.java]: If possible, could you remove this extra change? If not, no big deal.

Comment (TheIcyOne) [inline src/main/java/am2/blocks/BlockManaBattery.java]: How would I go about doing that? A little inexperienced with GitHub.

File: src/main/java/am2/blocks/BlockManaBattery.java [hunk 2]
```diff
@@ -134,6 +145,7 @@ public void onBlockPreDestroy(World par1World, int par2, int par3, int par4, int
 
 	private void destroy(World world, int i, int j, int k){
 		TileEntityManaBattery te = getTileEntity(world, i, j, k);
+
```
Comment (jjtParadox) [inline src/main/java/am2/blocks/BlockManaBattery.java]: Close! There's still just an extra line here.

**Comment by jjtParadox** [comment]
Has this been tested to be working in singleplayer and multiplayer?

**Comment by TheIcyOne** [comment]
I assumed multiplayer would act the same way as singleplayer, but I will test with multiplayer now.

**Comment by TheIcyOne** [comment]
Yep, everything's working fine in a multiplayer server too.

**Comment by TheIcyOne** [comment]
Actually, I did just come across an unusual quirk in that harvesting the battery with a pickaxe in multiplayer causes it to retain its power when the block provided is placed, whilst it loses power when broken with a pick in singleplayer.
I assume this is because harvestBlock is called at the point the block is broken, at which point a multiplayer server still retains the tile entity's data but the singleplayer server has, for some reason, already deleted it. Not sure how to unify the two, but one possible solution would be to force the block to drop without NBT if it is not picked up with the sneak+right click. Going to try that now.

**Comment by jjtParadox** [comment]
@TheIcyOne how is this PR looking? Any luck getting singleplayer and multiplayer working well?

**Comment by TheIcyOne** [comment]
Exams have gotten in the way a little, recently, but I'm kind of unhappy with the solution I came up with here, even though it is functional. Once my exams are over I'll come back and put a proper fix in, probably involving the Crystal Wrench.

**Comment by TheIcyOne** [comment]
Just going to close this for now, until I get back to working on it.

**Pull request closed without merging**